### DISCUSSION TITLE
SBOM (Software Bill of Materials)

### DISCUSSION THREAD

**Discussion by jase88** [nrwl/nx#20343] [General]
Hey,

Has anyone found a good and easy solution to generate sboms for apps within a nx monorepo? Couldn't find any plugin / generator.

We previously used `@cyclonedx/cyclonedx-npm` within a custom node script. But within a monorepo ideally only the dependencies of the selected app are considered. Any recommendations or experiences?

**Comment by GraceG7** [comment]
Hope this helps

https://medium.com/@gracegan_7/enhance-application-security-with-sbom-2cddd2edb6d0

**Comment by DanielRose** [comment]
Current versions of npm contains a command for generating a SBOM: [npm-sbom](https://docs.npmjs.com/cli/v11/commands/npm-sbom). You can run it via the [nx:run-commands](https://nx.dev/reference/core-api/nx/executors/run-commands) executor. For example in your project.json:

```json
{
  "targets": {
    "sbom": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm sbom --sbom-format spdx > sbom.json"
      }
    }
  }
}
```
